Sprint_6

SuslikCore opened the pull request.

Commit:
add jacoco report

okryukova commented on src/main/java/com/example/Lion.java:
```diff
@@ -4,9 +4,13 @@
 public class Lion {
 
+    /// убрал экзэмляр класса feline и заменил ее на приватное поле
+    private final Feline feline;
```
✅Отлично.  Правильно применил принцип инъекции зависимостей

okryukova commented on src/test/java/AnimalTests/ParameterizedAnimalTest.java:
```diff
@@ -0,0 +1,39 @@
+package AnimalTests;
```
⛔️Нужно исправить. Необходимо исправить наименование пакетов в соответствии с правилами нейминга. Пакеты — допустимы только строчные буквы. В случае, если слов несколько, их следует разделять точкой. Префикс должен быть одним из доменных имен верхнего уровня, таких как com, ru и т.д. Наименование пакета должно отражать суть содержимого. https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

okryukova commented on src/test/java/AnimalTests/ParameterizedAnimalTest.java:
```diff
@@ -0,0 +1,39 @@
+        List<String> expectedResult = foodType;
+
+        List<String> actualResult = animal.getFood(animalType);
+        System.out.printf("actualResult = %s%n", actualResult);
```
⛔️Нужно исправить. Вывод в консоль особо не нужен. Автотесты должны быть такими, чтобы не было нужды смотреть логи. Всё должно быть понятно по результату прохождения. Исправь, пожалуйста, этот момент во всем классе

okryukova commented on src/test/java/AnimalTests/SingleAnimalTests.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Test
+    public void getAnExceptionMessageTest() {
+        try {
```
⛔️Нужно исправить. В Junit4 есть метод assertThrows и @test(expected = Exception.class) для проверки исключений, try-cath не лучшая практика в тестах. Исправь, пожалуйста, этот момент во всем классе

Commit:
- changed packages' names
- deleted unnecessary printing in the console
- changed try-catch block to @Test(expected = Exception.class)

okryukova commented on src/test/java/animal/SingleAnimalTests.java:
```diff
@@ -0,0 +1,23 @@
+    @Test(expected = Exception.class)
+    public void getAnExceptionMessageTest() throws Exception {
+            Animal animal = new Animal();
+            animal.getFood("asd");
```
⚠️Можно улучшить. Можно проверить также текст исключения assertThrows
